lib/utils.pm

```diff
@@ -831,39 +834,128 @@ sub minimal_patch_system {
     }
 }
 
-=head2 zypper_search
+=head2 parse_zypper_table
 
- zypper_search($search_params);
+ parse_zypper_table($table, $fields);
 
-Run zypper search with given command line arguments and parse the output into
-an array of hashes.
+Parse ASCII table generated by commands such as C<zypper search> into an array
+of hashes. Pass the ASCII table in C<$table> as a single string. Pass an array
+of hash keys in C<$fields>. Keys will be mapped to columns by position.
```

Comment:
It's sad that the machine format of zypper it's only a wrapper for the text output of it :/

Reply:
Zypper also has `--xmlout` but unfortunately it's too broken. `zypper --xmlout search -s` does not report package retractions, `zypper --xmlout patches` prints the usual ASCII table wrapped in XML tags... Even if it gets fixed, we'll probably have to wait until everything before 12SP5 and 15SP4 reaches EOL.